File: solutions/go/armstrong-numbers/2/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

// IsNumber checks if a number is an Armstrong number.
// An Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
// For example, 153 is an Armstrong number because 1^3 + 5^3 + 3^3 = 153.
func IsNumber(n int) bool {
	var result int

	ns := strconv.Itoa(n)
	ndigits := len(ns)

	for _, c := range ns {
		num, _ := strconv.Atoi(string(c))
		result += int(math.Pow(float64(num), float64(ndigits)))
	}

	return result == n
}

File: solutions/c/armstrong-numbers/3/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include "math.h"

int total_digits(int num) {
    int result = 0;
    do {
        result += 1;
        num = num / 10;
    } while (num > 0);
    return result;
}

bool is_armstrong_number(int num) {
    int orig = num;
    int sum = 0;
    int d = 0;

    int n = total_digits(num);

    while (num > 0) {
        d = num % 10;
        sum += pow(d, n);
        num = num / 10;
    }

    return sum == orig;
}